src/datasets.js

```diff
@@ -322,13 +324,25 @@ export default {
         const parseResponse = (response) => {
             const contentType = response.headers['content-type'];
             const wrappedItems = wrapArray(response);
-            if (!disableBodyParser) wrappedItems.items = parseBody(wrappedItems.items, contentType);
+            try {
+                if (!disableBodyParser) wrappedItems.items = parseBody(wrappedItems.items, contentType);
+            } catch (e) {
```

Comment:
We should not be catching any error here because other error that may potentially happen is that JSON is too large to be parsed and in that case we don't want to retry.

Reply:
But we should definitely handle the "Unexpected end of JSON input" error this way

Reply:
So I can catch this particular error based on its message and rethrow the Retryable error otherwise I will throw the original one. Am I right?

Reply:
Yes, exactly we want to solve "Unexpected end of JSON input" error only because in a case of other error retry won't help